Lease timestamp not updated for home server when recoveryGroups and tran logs in a database is configured and database outage > couple of seconds occurs

**Describe the bug**  
When using a transaction recoveryGroup with the recovery logs in a database, if a database outage occurs, the server's lease renewal process's retry mechanism for its own lease does not perform apply the correct retry algorithm , using instead a lightweight retry mechanism thath gives up after a couple of seconds.  If that mechanism fails then the app server no longer updates its lease timestamp and other app servers in the recovery group will attempt peer recovery.  That peer recovery attempt will fail, because the logs heart beat timestamp will be current (that uses a different/the correct retry algoritm).


**Steps to Reproduce**  
Temporary (short duration of say 10 second) database outage results in peer app servers generating WTRN0108I: HADB Peer Recovery failed... messages

**Expected behavior**  
The app server should keep updating its lease timestamp and the peer recovery should not be attempted (at least not multiple times)

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.9 (should be for any version from 23.0.0.5 or versions with internal property enableHADBPeerLocking=true)
 - Affected feature(s) transactions

**Additional context**  
Add any other context about the problem here.

Additional call stacks for retry used:

Stack Dump = com.ibm.db2.jcc.am.ClientRerouteException: [jcc][t4][2027][11212][4.31.10] A connection failed but has been re-established. Special register settings have been replayed if necessary. Host name or IP address of the connection: 166.9.30.67. Service name or port number of the connection: 31,305.
Reason code: 1. Failure code: 2, Error code: zero byte read from the server.. ERRORCODE=-4498, SQLSTATE=08506
	at com.ibm.db2.jcc.am.b7.a(b7.java:322)
	at com.ibm.db2.jcc.am.b7.d(b7.java:454)
	at com.ibm.db2.jcc.t4.a.a(a.java:662)
	at com.ibm.db2.jcc.t4.a.a(a.java:619)
	at com.ibm.db2.jcc.t4.a.b(a.java:1185)
	at com.ibm.db2.jcc.t4.a.q(a.java:1043)
	at com.ibm.db2.jcc.t4.r.a(r.java:216)
	at com.ibm.db2.jcc.t4.a.a(a.java:1037)
	at com.ibm.db2.jcc.t4.y.b(y.java:312)
	at com.ibm.db2.jcc.t4.y.c(y.java:342)
	at com.ibm.db2.jcc.t4.y.c(y.java:455)
	at com.ibm.db2.jcc.t4.y.v(y.java:1230)
	at com.ibm.db2.jcc.t4.z.c(z.java:178)
	at com.ibm.db2.jcc.t4.n.c(n.java:42)
	at com.ibm.db2.jcc.t4.b.readLocalRollback_(b.java:3227)
	at com.ibm.db2.jcc.am.Connection.readRollback(Connection.java:1483)
	at com.ibm.db2.jcc.am.Connection.flowRollback(Connection.java:1444)
	at com.ibm.db2.jcc.t4.b.flowRollback(b.java:6382)
	at com.ibm.db2.jcc.am.Connection.rollbackX(Connection.java:1419)
	at com.ibm.db2.jcc.am.Connection.rollback(Connection.java:1407)
	at com.ibm.db2.jcc.am.du.rollback(du.java:180)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcConnection.rollback(WSJdbcConnection.java:2612)
	at com.ibm.ws.recoverylog.custom.jdbc.impl.SQLRetry.retryAfterSQLException(SQLRetry.java:176)
	at com.ibm.ws.recoverylog.custom.jdbc.impl.SQLSharedServerLeaseLog.claimPeerLeaseForRecovery(SQLSharedServerLeaseLog.java:1402)
	at com.ibm.tx.jta.impl.TxRecoveryAgentImpl.claimPeerLeaseForRecovery(TxRecoveryAgentImpl.java:821)
	at com.ibm.ws.recoverylog.spi.RecoveryDirectorImpl.peerRecoverServers(RecoveryDirectorImpl.java:1647)
	at com.ibm.tx.jta.impl.LeaseTimeoutManager$LeaseChecker.alarm(LeaseTimeoutManager.java:157)
	at com.ibm.tx.jta.util.alarm.AlarmManagerImpl$AlarmListenerWrapper.run(AlarmManagerImpl.java:85)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:191)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.lang.Thread.run(Thread.java:825)

Stack Dump = com.ibm.ws.recoverylog.spi.InternalLogException: com.ibm.db2.jcc.am.SqlException: The statement or command failed because it requires functionality that is not supported on a read-enabled HADR standby database. Reason code = "5".. SQLCODE=-1773, SQLSTATE=     , DRIVER=4.31.10
	at com.ibm.ws.recoverylog.custom.jdbc.impl.SQLSharedServerLeaseLog.claimPeerLeaseForRecovery(SQLSharedServerLeaseLog.java:1419)
	at com.ibm.tx.jta.impl.TxRecoveryAgentImpl.claimPeerLeaseForRecovery(TxRecoveryAgentImpl.java:821)
	at com.ibm.ws.recoverylog.spi.RecoveryDirectorImpl.peerRecoverServers(RecoveryDirectorImpl.java:1647)
	at com.ibm.tx.jta.impl.LeaseTimeoutManager$LeaseChecker.alarm(LeaseTimeoutManager.java:157)
	at com.ibm.tx.jta.util.alarm.AlarmManagerImpl$AlarmListenerWrapper.run(AlarmManagerImpl.java:85)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:191)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.lang.Thread.run(Thread.java:825)
Caused by: com.ibm.db2.jcc.am.SqlException: The statement or command failed because it requires functionality that is not supported on a read-enabled HADR standby database. Reason code = "5".. SQLCODE=-1773, SQLSTATE=     , DRIVER=4.31.10
	at com.ibm.db2.jcc.am.b7.a(b7.java:815)
	at com.ibm.db2.jcc.am.b7.a(b7.java:66)
	at com.ibm.db2.jcc.am.b7.a(b7.java:140)
	at com.ibm.db2.jcc.am.k_.b(k_.java:2491)
	at com.ibm.db2.jcc.am.k_.c(k_.java:2472)
	at com.ibm.db2.jcc.t4.ab.n(ab.java:917)
	at com.ibm.db2.jcc.t4.ab.a(ab.java:118)
	at com.ibm.db2.jcc.t4.p.a(p.java:50)
	at com.ibm.db2.jcc.t4.aw.b(aw.java:220)
	at com.ibm.db2.jcc.am.la.bm(la.java:3632)
	at com.ibm.db2.jcc.am.la.a(la.java:4677)
	at com.ibm.db2.jcc.am.la.b(la.java:4215)
	at com.ibm.db2.jcc.am.la.be(la.java:832)
	at com.ibm.db2.jcc.t4.j.be(j.java:198)
	at com.ibm.db2.jcc.am.la.executeUpdate(la.java:806)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcPreparedStatement.executeUpdate(WSJdbcPreparedStatement.java:522)
	at com.ibm.ws.recoverylog.custom.jdbc.impl.SQLSharedServerLeaseLog.claimPeerLeaseFromTable(SQLSharedServerLeaseLog.java:1503)
	at com.ibm.ws.recoverylog.custom.jdbc.impl.SQLSharedServerLeaseLog.claimPeerLeaseForRecovery(SQLSharedServerLeaseLog.java:1322)
	... 11 more

--- a/dev/com.ibm.rls.jdbc/src/com/ibm/ws/recoverylog/custom/jdbc/impl/SQLSharedServerLeaseLog.java
+++ b/dev/com.ibm.rls.jdbc/src/com/ibm/ws/recoverylog/custom/jdbc/impl/SQLSharedServerLeaseLog.java
@@ -503,8 +503,8 @@ public class SQLSharedServerLeaseLog extends LeaseLogImpl implements SharedServe
                     updateServerLeaseRetry.setNonTransientException(currentSqlEx);
                     // The following method will reset "nonTransientException" if it cannot recover
                     if (_sqlTransientErrorHandlingEnabled) {
-                        failAndReport = updateServerLeaseRetry.retryAfterSQLException(this, currentSqlEx, SQLRetry.getLightweightRetryAttempts(),
-                                                                                      SQLRetry.getLightweightRetrySleepTime());
+                        failAndReport = updateServerLeaseRetry.retryAfterSQLException(this, currentSqlEx, SQLRetry.getTransientRetryAttempts(),
+                                                                                      SQLRetry.getTransientRetrySleepTime());
 
                         if (failAndReport)
                             nonTransientException = updateServerLeaseRetry.getNonTransientException();

--- a/dev/com.ibm.tx.jta/src/com/ibm/tx/jta/impl/LeaseTimeoutManager.java
+++ b/dev/com.ibm.tx.jta/src/com/ibm/tx/jta/impl/LeaseTimeoutManager.java
@@ -1,12 +1,12 @@
 package com.ibm.tx.jta.impl;
 
 /*******************************************************************************
- * Copyright (c) 2002, 2022 IBM Corporation and others.
+ * Copyright (c) 2002, 2023 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-2.0/
- * 
+ *
  * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
@@ -62,10 +62,9 @@ public class LeaseTimeoutManager {
             _leaseLog = leaseLog;
 
             // Renew lease
-            // Disable lease checking if we failed to renew our own lease;
-            if (renewLease()) {
-                schedule(delay);
-            }
+            renewLease();
+            // Schedule the next renewal irrespective of the success or failure of the last renewal attempt
+            schedule(delay);
         }
 
         /*
@@ -78,7 +77,10 @@ public class LeaseTimeoutManager {
             if (tc.isDebugEnabled())
                 Tr.debug(tc, "LeaseRenewal", _recoveryIdentity, _recoveryGroup);
 
-            if (!FrameworkState.isStopping() && renewLease()) {
+            if (!FrameworkState.isStopping()) {
+                // Renew lease
+                renewLease();
+                // Schedule the next renewal irrespective of the success or failure of the last renewal attempt
                 schedule((int) delay);
             }
         }
@@ -177,6 +179,7 @@ public class LeaseTimeoutManager {
                 _alarm = null;
             }
         }
+
     }
 
     /**

